GVFS/GVFS/CommandLine/IncludeVerb.cs

```diff
@@ -0,0 +1,173 @@
+                    {
+                        IncludedFolderTable includedFolderTable = new IncludedFolderTable(database);
+
+                        if (this.List)
+                        {
+                            List<string> directories = includedFolderTable.GetAll();
+                            foreach (string directory in directories)
+                            {
+                                this.Output.WriteLine(directory);
+                            }
+
+                            return;
+                        }
+
+                        // Make sure there is a clean git status before allowing inclusions to change
+                        this.CheckGitStatus(tracer, enlistment);
+
+                        if (!string.IsNullOrEmpty(this.Remove))
+                        {
+                            foreach (string directoryPath in this.Remove.Split(new[] { ';' }, StringSplitOptions.RemoveEmptyEntries))
+                            {
+                                includedFolderTable.Remove(directoryPath);
+                            }
+                        }
+
+                        if (!string.IsNullOrEmpty(this.Add))
+                        {
+                            foreach (string directoryPath in this.Add.Split(new[] { ';' }, StringSplitOptions.RemoveEmptyEntries))
+                            {
+                                includedFolderTable.Add(directoryPath);
+                            }
+                        }
+                    }
+
+                    // Force a projection update
+                    this.ForceProjectionChange(tracer, enlistment);
+                }
+                catch (Exception e)
+                {
+                    tracer.RelatedError(e.Message);
+                }
+            }
+        }
+
+        private void ForceProjectionChange(ITracer tracer, GVFSEnlistment enlistment)
+        {
+            string errorMessage = null;
+
+            if (!this.ShowStatusWhileRunning(
+                () =>
+                {
+                    GitProcess git = new GitProcess(enlistment);
+                    GitProcess.Result checkoutResult = git.ForceCheckout("HEAD");
+
+                    errorMessage = checkoutResult.Errors;
+                    return checkoutResult.ExitCodeIsSuccess;
+                },
+                "Forcing a projection change",
+                suppressGvfsLogMessage: true))
+            {
+                this.WriteMessage(tracer, "Failed to change projection: " + errorMessage);
+            }
+        }
+
+        private void CheckGitStatus(ITracer tracer, GVFSEnlistment enlistment)
+        {
+            GitProcess.Result statusResult = null;
+            if (!this.ShowStatusWhileRunning(
+                () =>
+                {
+                    GitProcess git = new GitProcess(enlistment);
+                    statusResult = git.Status(allowObjectDownloads: false, useStatusCache: false);

```

Comment:
I noticed that our status invocation doesn't override status.showuntrackedfiles, so there is a edge case here where a user has untracked file(s) but a naked `git status` won't show them, reporting working tree clean. That might lead to some wonkiness, especially if (going even more edge-casey here) the file has the same name as a directory that is being added to the projection.

Maybe we should either always force that config setting to true when we call status, if that's more correct for all usages of `Status()` in our codebase,  or we could plumb in an option to the `Status()` method.

Reply:
Thanks I will test this more.  There is also the question of ignored files and how we would handle them but like you said these are really on the edge of doing some weird things but we should definitely try to handle them as best we can.  BTW the status in VFSForGit is returning:
```
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit (use -u to show untracked files)
```
Which is weird because running `git status` from the command line gives you
```
C:\gvfs_repos\ft2\src>git status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
```